piperider_cli/assertion_engine/assertion.py

```diff
@@ -217,12 +226,13 @@ def load_assertions(self):
 
         # Load assertion
         for t in self.assertions_content:
-            for ta in self.assertions_content[t].get('tests', []):
-                self.assertions.append(AssertionContext(t, None, ta.get('name'), self.assertions_content))
-            for c in self.assertions_content[t].get('columns', {}):
-                for ca in self.assertions_content[t]['columns'][c].get('tests', []):
-                    self.assertions.append(AssertionContext(t, c, ca.get('name'), self.assertions_content))
-        pass
+            # only append specified table's assertions
+            if t in set(table.lower() for table in list(self.profiler.metadata.tables)):
```

Comment:
`set(table.lower() for table in list(self.profiler.metadata.tables))` can be extracted to line 226. The result of the `set` won't be changed during the for-loop.

Reply:
Thanks! It should be enhanced.